### PR TITLE
Torinnayak/rep 19 onboarding fetch data from rails

### PR DESCRIPTION
**Pull request by torinnayak1**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

Onboarding Page pulls the Partner options from the database, and the Finish button updates the partner_id of the current driver (currently just Driver 1).

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

rails server -b 0.0.0.0 -p 3000

won't work until pr created/merged on business repo sorry

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

Need to update once Stephen's PR is merged on the backed to use the current driver that is signed in, not a hard coded driver.

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @rachaelch3n

### CODE REVIEW

**Review by jeannineruiz — APPROVED** [review]
LGTM ✅

File: src/app/onboarding/index.tsx [hunk 1]
```diff
@@ -1,21 +1,32 @@
...
+        console.error('Error fetching partners:', err);
+      }
+    })();
+  }, []);
+
+  useEffect(() => {
+    setItems(partners.map(npo => ({ label: npo[1], value: npo[0] })));
+  }, [partners]);
+
+  const handleUpdatePress = () => async () => {
```
Comment (stephenhungg) [inline src/app/onboarding/index.tsx:27]: Bug: incorrect async handler in onboarding/index.tsx (lines 27-29)
Problem: Returns a function that returns a Promise. Fix: make it a direct async function or call it correctly.

File: src/app/onboarding/index.tsx [hunk 2]
```diff
@@ -1,21 +1,32 @@
...
-  ];
-
+  const [partners, setPartners] = useState([]);
   const [open, setOpen] = useState(false);
-  const [selectedNPOId, setSelectedNPOId] = useState(null);
-  const [items, setItems] = useState(
-    MOCK_NPOS.map(npo => ({ label: npo.name, value: npo.id })),
-  );
+  const [selectedNPOId, setSelectedNPOId] = useState(0);
+  const [items, setItems] = useState([{}]);
```
Comment (stephenhungg) [inline src/app/onboarding/index.tsx:10]: Invalid initial state (line 10)
Fix: Use useState([]).

File: api/config.ts [hunk 3]
```diff
@@ -68,3 +71,47 @@ export const driverAPI = {
...
+    throw new Error(`Failed to fetch partners: ${text}`);
+  }
+
+  const json = await response.json();
+  return json;
+};
+
+export async function updateDriverPartner(selectedNPOId: number) {
+  try {
+    const response = await fetch(`${BASE_URL}${API_ENDPOINTS.DRIVERS}/1`, {
```
Comment (stephenhungg) [inline api/config.ts:89]: Hardcoded driver ID (config.ts line 89)
Note: PR mentions this needs to use the current logged-in driver.

**Review by stephenhungg — APPROVED** [review]
Overall: Good, but fix the critical bugs before merging, especially the async handler.
